src/mds/MDSMap.cc

```diff
@@ -863,8 +870,24 @@ void MDSMap::decode(bufferlist::const_iterator& p)
   } else if (ev > 14) {
     decode(min_compat_client, p);
   }
-
+  if (decode_ev)
+    decode(encode_ev, p);
```

Comment:
if (struct_v >= 6)
      decode(decode_ev, p);
else
      decode_ev = false;

Reply:
@ukernel We are using  `decode_ev` before that - just below `// kclient ignores everything from here` for doing a `ceph_assert`

Reply:
@ukernel Done, as per your irc reply

Reply:
this code is less clear than I suggested